internal/csi-common/utils.go

```diff
@@ -320,3 +320,46 @@ func IsBlockMultiNode(caps []*csi.VolumeCapability) (bool, bool) {
 
 	return isBlock, isMultiNode
 }
+
+// IsMultiWriterBlockRWOFile validates the volume capability slice against the access modes and access type.
```

Comment:
This function is horrible, it returns **FOUR** booleans. It would be much easier to read when the function is split in multiple functions, each returning their own `isBlock()`, `isMultiWriter()` and so on. There are very few capabilities in the slice that is passed to this function, there is no need to optimize for looping fewer times. Just have the caller call `isMultiWriter()` where it needs. It will improve the maintainability _so much_.

Reply:
Done. splitted into different functions.